packages/interactive_media_ads/android/src/main/kotlin/dev/flutter/packages/interactive_media_ads/ProxyApiRegistrar.kt

```diff
@@ -0,0 +1,124 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package dev.flutter.packages.interactive_media_ads
+
+import android.content.Context
+import android.os.Handler
+import android.os.Looper
+import io.flutter.plugin.common.BinaryMessenger
+
+open class ProxyApiRegistrar(binaryMessenger: BinaryMessenger, var context: Context) :
+    PigeonProxyApiRegistrar(binaryMessenger) {
+
+  // Added to be overriden for tests. The test implementation calls `callback` immediately, instead
+  // of waiting for the main thread to run it.
+  internal open fun runOnMainThread(callback: Runnable) {
+    Handler(Looper.getMainLooper()).post { callback.run() }
+  }
```

Comment:
This was added to the registrar to make it easier to handle and test callback methods on the correct thread.